src/components/container/BloodPressureVisualization/BloodPressureVisualization.tsx

```diff
@@ -0,0 +1,257 @@
+import "./BloodPressureVisualization.css";
+import { useEffect, useState } from "react";
+import { debounce } from "lodash";
+import MyDataHelps, { Guid, SurveyAnswer, SurveyAnswersPage, SurveyAnswersQuery } from "@careevolution/mydatahelps-js";
+import plus from './plus.svg';
+import React from "react";
+import { Button, DateRangeNavigator, LoadingIndicator } from "../../presentational";
+import DumbbellChart from "../../presentational/DumbbellChart";
+import { Dumbbell, ClosedInterval, Axis, DataPoint, DumbbellClass } from "../../presentational/DumbbellChart/DumbbellChart";
+import { previewSurveyData } from './BloodPressureVisualization.previewdata'
+import { addDays, format, isEqual, isMonday, nextSunday, parseISO, previousMonday, startOfDay } from "date-fns";
+
+export type DataFaucet = "PreviewSurvey" | "Survey";
+
+export interface BloodPressureVisualizationProps {
+    source : DataFaucet,
+    surveyName?: string,
+    stepIdentifier? : string,
+    dateOfResultId? : string,
+    systolicResultId? : string,
+    diastolicResultId? : string
```

Comment:
I would make this a lot less tied to surveys or any particular data source; instead I think this viz should take a set of "BloodPressureDataProviders".  There would just be 1 BloodPressureDataProvider for now - "SurveyBloodPressureDataProvider" - but it would enable us to add more in the future for Apple Health etc.

The Viz would just invoke the data provider; so all the querying of survey results would move out of this component and into the "SurveyBloodPressureDataProvider"

See for instance how I did this with "Daily data" providers: https://github.com/CareEvolution/MyDataHelpsUI/tree/main/src/helpers/daily-data-providers
https://github.com/CareEvolution/MyDataHelpsUI/pull/4